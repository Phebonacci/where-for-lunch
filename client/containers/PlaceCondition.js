import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import Condition from 'components/Condition/Condition';
import conditionActions from 'actions/conditionActions';

const mapStateToProps = state => ({
  condition: state.condition,
  pricingLevels: state.pricingLevels,
});

const mapDispatchToProps = dispatch =>
  bindActionCreators({
    setRadius: conditionActions.setRadius,
    setPriceSelection: conditionActions.setPriceSelection,
  }, dispatch);

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(Condition);
