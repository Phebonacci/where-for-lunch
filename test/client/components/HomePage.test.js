import React from 'react';
import { render } from 'enzyme';
import { HomePage } from 'client/containers/HomePage';
import renderer from 'react-test-renderer';

jest.mock('components/Place/Place', () => 'Place');
jest.mock('containers/PlaceCondition', () => 'PlaceCondition');

test('With Enzyme, HomePage component disables out the search Button when no latitude is fetched', () => {
  const condition = { radius: 500 };
  const wrapper = render(
    <HomePage condition={condition} />,
  );
  const disabledButton = wrapper.find('.homepageClick.disabled');
  expect(disabledButton.length).toBe(1);
});

test('With Jest snapshot, HomePage component disables and greys out the search Button when no latitude is fetched', () => {
  const condition = { radius: 500 };
  const homePageComponent = renderer
    .create(<HomePage condition={condition} />)
    .toJSON();
  expect(homePageComponent).toMatchSnapshot();
});

test('With Jest snapshot, HomePage component enables the search Button when a latitude is fetched', () => {
  const condition = { radius: 500, latitude: 7.1 };
  const homePageComponent = renderer
    .create(<HomePage condition={condition} />)
    .toJSON();
  expect(homePageComponent).toMatchSnapshot();
});
