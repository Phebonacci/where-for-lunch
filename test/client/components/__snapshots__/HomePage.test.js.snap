// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Jest snapshot, HomePage component disables and greys out the search Button when no latitude is fetched 1`] = `
<div
  className="homePageWrapper"
>
  <Place />
  <div
    className="searchWrapper"
  >
    <div
      className="root"
    >
      <span>
        radius:
      </span>
      <div
        className="root"
      >
        <input
          defaultValue={500}
          onBlur={[Function]}
          type="text"
        />
      </div>
      <span>
        meters
      </span>
    </div>
    <button
      className="root homepageClick disabled"
      disabled={true}
      onClick={[Function]}
    />
  </div>
</div>
`;

exports[`With Jest snapshot, HomePage component enables the search Button when a latitude is fetched 1`] = `
<div
  className="homePageWrapper"
>
  <Place />
  <div
    className="searchWrapper"
  >
    <div
      className="root"
    >
      <span>
        radius:
      </span>
      <div
        className="root"
      >
        <input
          defaultValue={500}
          onBlur={[Function]}
          type="text"
        />
      </div>
      <span>
        meters
      </span>
    </div>
    <button
      className="root homepageClick disabled"
      disabled={true}
      onClick={[Function]}
    />
  </div>
</div>
`;
